libs/application/ui-shell/src/components/ScreenFooter.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { FC } from 'react'
+import {
+  Box,
+  ButtonDeprecated as Button,
+  GridColumn,
+} from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import * as styles from './Screen.treat'
+import { FormModes, StaticText, SubmitField } from '@island.is/application/core'
+
+interface FooterProps {
+  mode?: FormModes
+  activeScreenIndex: number
+  numberOfScreens: number
+  goBack: () => void
+  submitField?: SubmitField
+  loading: boolean
+  canProceed: boolean
+}
+
+export const ScreenFooter: FC<FooterProps> = ({
+  activeScreenIndex,
+  canProceed,
+  goBack,
+  loading,
+  mode,
+  numberOfScreens,
+  submitField,
+}) => {
+  const { formatMessage } = useLocale()
+  const hasSubmitField = submitField !== undefined
+  const isLastScreen = activeScreenIndex === numberOfScreens - 1
+  const showGoBack = activeScreenIndex > 0
+  if (
+    isLastScreen ||
+    (mode !== FormModes.REVIEW && mode !== FormModes.APPLYING)
+  ) {
+    return null
+  }
+
+  function getSubmitButtonText(): StaticText {
+    if (submitField?.placement === 'footer') {
+      const { actions } = submitField
+      if (actions.length === 1) {
+        return actions[0].name
+      }
+    }
+    return {
```

Comment:
I think you might need to define this message with
` const submitText = defineMessage({
      id: 'application.system:button.submit',
      defaultMessage: 'Submit',
      description: 'Submit button text',
})
`
and then return that for the extract script to work, not sure though, but you could try and see if this message appears in the messages.json in the localization lib.

Reply:
yeah, I moved this just from the render function really. I think we need to take a look at these strings in a future PR when we will apply stricter rules on translations before going live :)